mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java

```diff
@@ -190,6 +190,15 @@
     MetaData getMetaData(boolean resetRecent, MailboxSession mailboxSession, MessageManager.MetaData.FetchGroup fetchGroup) throws MailboxException;
 
     /**
+     * Offers direct access to UIDs. This allows per implementation optimisation for retrieving the whole UIDs of a mailbox.
+     *
+     * @param session MailboxSession for the user performing the request.
+     * @return Iterator of MessageUids in this mailbox. They are not necessarily sorted.
+     * @throws MailboxException
+     */
+    Iterator<MessageUid> getUids(MailboxSession session) throws MailboxException;
```

Comment:
I don't agree : you will have to ways to retrieve the same information with the same interface, you put the complexity on the caller.